examples/arithmetics/src/language-server/arithmetics.langium

```diff
@@ -20,19 +20,9 @@ Evaluation:
     expression=Expression ';';
 
 Expression:
-    Addition;
+    BinaryExpression;
 
-Addition infers Expression:
-    Multiplication ({infer BinaryExpression.left=current} operator=('+' | '-') right=Multiplication)*;
-
-Multiplication infers Expression:
-    Exponentiation ({infer BinaryExpression.left=current} operator=('*' | '/') right=Exponentiation)*;
-
-Exponentiation infers Expression:
-    Modulo ({infer BinaryExpression.left=current} operator='^' right=Modulo)*;
-
-Modulo infers Expression:
-    PrimaryExpression ({infer BinaryExpression.left=current} operator='%' right=PrimaryExpression)*;
+infix BinaryExpression on PrimaryExpression: '%' > '^' > '*' | '/' > '+' | '-';
```

Comment:
Normally the precedence of > is greater than the one of |. It reads a bit unnatural.
Adding parentheses could be an option to enumerate all alternatives.
Or maybe a equal sign instead of the |.
WDYT?

Reply:
And why not using the comma operator instead of '>' ?
```
'%' , '^' , '*' | '/' , '+' | '-';
```

Reply:
‚>‘ implies an order between the operators, that makes totally sense to me. Having a comma, you do not know what comes first: low or high precedence. But you can assume some convention, true.

Reply:
You can mitigate this by writing each group in its own line:
```
infix BinaryExpression on PrimaryExpression:
    '%'
    > '^'
    > '*' | '/'
    > '+' | '-';
```
This is how I'd format such expressions for readability.

Reply:
@msujew shall we change the formatting here to give a better readable example?